# Masking arrays below some value

This function was created mainly for plotting purposes. It is sometimes useful to mask values that are too small to matter (sad, we know). This is specially useful when in a plot we only want big differences to show up. So we have created a small function that masks values in an array that are below some threshold that the user can define. Let's import obrero: 

In [1]:
# small hack to be able to import module without install
import os
import sys
sys.path.append(os.getcwd() + '/../')

import obrero

Now we read some data:

In [2]:
# read data
fname = 'data/ctl.nc'

# read as data array
da = obrero.read_nc(fname, 'pr')

# convert units
da.convert_units('mm day-1')

And say we don't want small values of precipitation to show up in a plot. Maybe less than 0.1 mm day$^{-1}$ is very little to provide any useful information, so we want to mask values below this threshold for some later plotting. We do so using the function `mask_below()`, part of the `obrero.utils` module:

In [8]:
mda = obrero.mask_below(da, 0.1)
mda

<xarray.DataArray 'pr' (time: 72, latitude: 32, longitude: 64)>
array([[[0.70923 , 0.777796, ..., 0.540069, 0.629775],
        [0.974657, 1.099068, ..., 0.706435, 0.826919],
        ...,
        [0.227743, 0.123709, ..., 0.353112, 0.311241],
        [0.373475, 0.427707, ..., 0.332196, 0.348443]],

       [[0.186415, 0.181961, ..., 0.169736, 0.178863],
        [1.689129, 1.527837, ..., 1.74003 , 1.793405],
        ...,
        [     nan,      nan, ...,      nan,      nan],
        [0.2091  , 0.248029, ..., 0.163378, 0.182041]],

       ...,

       [[0.555498, 0.599123, ..., 0.43743 , 0.504916],
        [1.976995, 1.71042 , ..., 2.458799, 2.274632],
        ...,
        [0.751534, 0.88062 , ..., 0.371274, 0.526058],
        [0.120024,      nan, ..., 0.159671, 0.138846]],

       [[0.625771, 0.617749, ..., 0.601025, 0.618503],
        [1.539145, 1.30133 , ..., 1.79069 , 1.720224],
        ...,
        [0.485342, 0.515671, ..., 0.27923 , 0.394253],
        [0.249841, 0.239115, ..., 0.2462

Notice there are some NaN values. If we want to count how many NaN values there are, we need to import numpy and count:

In [13]:
import numpy as np
np.isnan(mda.values).sum()

10397

So there are 10397 values that have been masked. **Know that this function uses the absolute values of the array. This means that negative values will also be masked, and that the threshold can only be a positive number**.